### DISCUSSION TITLE
API to fetch tokens programatically

### DISCUSSION THREAD

**Discussion by PrayagS** [ente-io/ente#1020] [Enhancements]
**Is your feature request related to a problem? Please describe.**

I've migrated to ente auth from Authy as many others. What I miss here is a Raycast extension that lets me fetch tokens.

That reduces the context switch between the desktop and the phone or even between the textbox and the desktop app for ente auth.

Authy has an extension like that built by the community: https://github.com/raycast/extensions/blob/main/extensions/authy/README.md

**Describe the solution you'd like**

Some kind of API that lets users fetch tokens.

I believe something like that should already be in place given that the desktop app is already being developed.

**Comment by ua741** [comment]
The underlying tech for Raycast extensions is very similar to current stack for Auth web.
https://developers.raycast.com/
> Powerful and familiar tooling: Extensions are built with TypeScript, React, and Node. Leverage npm's ecosystem to quickly build what you imagine.

If anyone is interested in building this extension, they can refer to the auth web code here. 
https://github.com/ente-io/ente/tree/main/web/apps/auth

**Comment by PrayagS** [comment]
Awesome. Thanks for sharing that.

I might hack around this during the weekend.

**Comment by vfonic** [comment]
Hey @PrayagS ! Did you have any luck building an extension? Thanks!

**Comment by CodeWithShreyans** [comment]
any updates? this is a major feature requirement for me

**Comment by dabeeeenster** [comment]
can we achieve this with the cli now?

**Comment by nktnet1** [comment]
There is now an Ente Auth integration in Raycast
- https://www.raycast.com/chkpwd/ente-auth
